### PR TITLE
Convert lib/utils/tty to typescript

### PR DESCRIPTION
**Pull request by Page-**
Change-type: patch

Resolves: #   
Change-type: major|minor|patch   
Depends-on: <url>   
See: <url>   

---
Please check the CONTRIBUTING.md file for relevant information and some
guidance. Keep in mind that the CLI is a cross-platform application that runs
on Windows, macOS and Linux. Tests will be automatically run by balena CI on
all three operating systems, but this will only help if you have added test
code that exercises the modified or added feature code.

Note that each commit message (currently only the first line) will be
automatically copied to the CHANGELOG.md file, so try writing it in a way
that describes the feature or fix for CLI users.

If there isn't a linked issue or if the linked issue doesn't quite match the
PR, please add a PR description to explain its purpose or the features that it
implements. Adding PR comments to blocks of code that aren't self explanatory
usually helps with the review process.

If the PR introduces security considerations or affects the development, build
or release process, please be sure to highlight this in the PR description.

### CODE REVIEW

File: lib/utils/tty.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	const hideCursor = () => stream.write('\u001B[?25l');
+
+	const showCursor = () => stream.write('\u001B[?25h');
+
+	const cursorUp = (rows: number = 0) => {
+		return stream.write(`\u001B[${rows}A`);
+	};
+
+	const cursorDown = (rows: number = 0) => {
+		return stream.write(`\u001B[${rows}B`);
```
Comment (pdcastro) [inline lib/utils/tty.ts]: NIT: `cursorUp` and `cursorDown` could be written in the same style as the other functions, without the `return` keyword.

**Comment by pdcastro** [comment]
Apparently I cannot [dismiss](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/dismissing-a-pull-request-review) the review approval I had given 2 months ago to this PR. Heads up that when [bulldozer (automerge bot)](https://www.flowdock.com/app/rulemotion/d-build-pipelines/threads/pmvO-nSO_Idz-Z1MDwP7z3zt3zR) is deployed, the PR could be automatically merged (?).

**Comment by Page-** [comment]
It shouldn't merge it due to the failed test @pdcastro, but either way once I rebase it again it should be good to merge anyway, just had other stuff on my plate to finish

**Comment by Page-** [comment]
Tested and looks good

**Comment by pdcastro** [comment]
> Tested and looks good

Currently the PR has 188 commits though... :-) Was it tested on top of the current master branch?

**Comment by Page-** [comment]
It was @pdcastro, if you check the commits they're all the commits from the master branch :P  I did push during a period github was having an incident so I'm guessing it bugged out when trying to reconcile the new branch commits vs the master commits and couldn't work out they were the same

**Comment by Page-** [comment]
@balena-ci retest

**Pull request merged**